bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -4,102 +4,163 @@ use core::fmt;
 use core::ops::{Div, Mul};
 
 use super::Weight;
-use crate::prelude::*;
 use crate::Amount;
 
 /// Represents fee rate.
 ///
 /// This is an integer newtype representing fee rate in `sat/kwu`. It provides protection against mixing
 /// up the types as well as basic formatting features.
 #[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
-#[cfg_attr(feature = "serde", serde(transparent))]
-pub struct FeeRate(u64);
+pub struct FeeRate {
+    per_kwu: Amount,
```

Comment:
Maybe this could just be an enum instead, like so:

```
enum FeeRate {
    per_wu(Amount),
    per_vb(Amount)
}
```
?

Reply:
Nah they are trivially equivalent just a factor of 4 difference..